kyo-data/shared/src/main/scala/kyo/Result.scala

```diff
@@ -7,21 +7,47 @@ import scala.reflect.ClassTag
 import scala.util.Try
 import scala.util.control.NonFatal
 
+/** Represents a result that can be either a success, a failure or an unexpected panic.
+  *
+  * @tparam E
+  *   The type of the error in case of failure
+  * @tparam A
+  *   The type of the value in case of success
+  */
 opaque type Result[+E, +A] >: (Success[A] | Error[E]) = Success[A] | Error[E]
 
+/** Companion object for Result type */
 object Result:
     import internal.*
 
     inline given [E, A](using inline ce: CanEqual[A, A]): CanEqual[Result[E, A], Result[E, A]] = CanEqual.derived
-    given [E, A]: CanEqual[Result[E, A], Panic]                                                = CanEqual.derived
-
+    inline given [E, A]: CanEqual[Result[E, A], Panic]                                         = CanEqual.derived
+
+    /** Creates a Result from an expression that might throw an exception.
+      *
+      * @param expr
+      *   The expression to evaluate
+      * @tparam A
+      *   The type of the successful result
+      * @return
+      *   A Result containing either the successful value or a Panic with the caught exception
+      */
     inline def apply[A](inline expr: => A): Result[Nothing, A] =
         try
             Success(expr)
         catch
             case ex => Panic(ex)
```

Comment:
`NonFatal`? Line 56 as well?

Reply:
I put the `NonFatal` check in `Panic.apply` to reduce the size of the inlined bytecode.

Reply:
Ah clever, I must have forgotten.